ssl/ech/ech_ssl_apis.c

```diff
@@ -184,7 +184,7 @@ int SSL_ech_get1_status(SSL *ssl, char **inner_sni, char **outer_sni)
             return SSL_ECH_STATUS_GREASE_ECH;
         return SSL_ECH_STATUS_GREASE;
     }
-    if (s->options & SSL_OP_ECH_GREASE)
+    if (s->options & SSL_OP_ECH_GREASE && s->ext.ech.attempted != 1)
```

Comment:
C operator issue  : Logical && has an higher priority than binary & 

you should write :
```C
if ((s->options & SSL_OP_ECH_GREASE) !=0 && s->ext.ech.attempted != 1)
```

Reply:
sure

Reply:
Mind you, not sure I agree that "Logical && has an higher priority than binary &" but no harm making the change anyway